### PR TITLE
Project Authentication week 20: Elle Frankenberg & Ida Sylvén

### PR DESCRIPTION
**Pull request by idasylven**
Link to our deployed project:
https://elegant-hopper-4950bc.netlify.app/
https://pingu-register-app.herokuapp.com/

Our reflections on how this week's project turned out:
We are very happy with our project! We had a super fun week working together, some stuggles with internet connection and a tiny headache since we had to restart or repo and forgot to an import bcrypt and crypto. It has been a great project to go back to for future reference. 

Things we'd like to have clarified or explained in more detail:
Nothing at this current stage, we just need to practice more :-)

### CODE REVIEW

File: frontend/src/components/Main.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      localStorage.removeItem('user')
+    })
+  }
+
+  return (
+    <div className="main-container">
+      <div className="backend-text-container">
+        <h1 className="backend-text">The back end!</h1>
+      </div>
+      <Link className="link-to-register" to="/">To Register we go!</Link>
```
Comment (IsabellaM5) [inline frontend/src/components/Main.js:34]: Do you need this Link here at all? 😊 If you're logged in you don't need to go to the register page.

File: frontend/src/components/Main.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      <img className="machines-illustration" src="./assets/backend.gif" alt="communicating machines illustration" />
```
Comment (IsabellaM5) [inline frontend/src/components/Main.js:35]: LOOOVE THE GIF 😍

File: frontend/src/components/Main.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        </button>
+      </div>
+    </div>
+  )
+}
+
+export default Main
+
+
+//
```
Comment (IsabellaM5) [inline frontend/src/components/Main.js:52]: 🧹🧹🧹

**Review by IsabellaM5 — APPROVED** [review]
Fantastic job this week Ida and Elle! You did all the blue level requirements and we absolutely loved the concept and end product! 😍

For specific comments/suggestions, see the code files 😊

//Isabella & Tobias 🐯🐯

File: README.md [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
+When we have more time we would like to investigate why the inital page load at times is a bit slow, perhaps our image content might be a bit too heavy. Initially we also wanted to use Styled Components, but decided to go for separate css-files as we both felt more comfortable with that approach.
+
+We are very happy with our visuals - and the code too!
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+To view this project live, please follow these links:
+
+Deployed site: https://elegant-hopper-4950bc.netlify.app/
```
Comment (KarolCislowski) [inline README.md:25]: To fix it you need to create `_redirects` file in the `public` directory of your react project and paste it inside:
``` 
/* /index.html 200
```

image

File: backend/server.js [hunk 5]
```diff
@@ -1,30 +1,117 @@
...
+    required: true,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString("hex"),
```
Comment (KarolCislowski) [inline backend/server.js:31]: nice model 👍🏻

File: frontend/src/components/Main.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  }, [accessToken, history])
+
+  const handleLogOut = () => {
+    batch(() => {
+      dispatch(user.actions.setUsername(null))
+      dispatch(user.actions.setAccessToken(null))
+
+      localStorage.removeItem('user')
+    })
```
Comment (KarolCislowski) [inline frontend/src/components/Main.js:26]: Nice 👍🏻

File: frontend/src/components/Register.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      .then(data => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUsername(data.username))
+            dispatch(user.actions.setAccessToken(data.accessToken))
+            dispatch(user.actions.setErrors(null))
+
+            localStorage.setItem('user', JSON.stringify({
+              username: data.username,
+              accessToken: data.accessToken
```
Comment (KarolCislowski) [inline frontend/src/components/Register.js:51]: Gret idea 👍🏻

File: frontend/src/reducers/user.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    setAccessToken: (store, action) => {
+      store.accessToken = action.payload
+    },
+    setErrors: (store, action) => {
+      store.errors = action.payload
+    },
+  }
+})  
+
+export default user
```
Comment (KarolCislowski) [inline frontend/src/reducers/user.js:31]: great reducer!

**Review by KarolCislowski — APPROVED** [review]
Great app with nice styling. I will approve it but please fix the Netlify error page issue-instructions below.

**Pull request closed without merging**